### PR TITLE
ENT-9363: Improved MacOS Support

### PR DESCRIPTION
**Pull request by craigcomstock**
- Find openssl libraries in a more dynamic way.
- Fixup unix_iface.c compiler warnings.

### CODE REVIEW

File: .github/workflows/macos_unit_tests.yml [hunk 1]
```diff
@@ -11,9 +11,9 @@ jobs:
       with:
         submodules: recursive
     - name: Install dependencies
-      run: brew install lmdb automake openssl@1.1 pcre
+      run: brew install lmdb automake openssl pcre
```
Comment (craigcomstock) [inline .github/workflows/macos_unit_tests.yml:14]: removing the `@1.1` has the effect in github of a no-op. openssl@3 is already installed in their image I guess

``` 
Warning: openssl@3 3.0.8 is already installed and up-to-date.
To reinstall 3.0.8, run:
  brew reinstall openssl@3
```

File: libenv/unix_iface.c [hunk 2]
```diff
@@ -711,13 +711,15 @@ static void FindV6InterfacesInfo(EvalContext *ctx, Rlist **interfaces, Rlist **h
...
 
+#ifndef NDEBUG
                 // We know there was more data, at least a colon:
                 assert(src_length == bytes_to_copy);
                 const size_t dst_length = src_length - 1;
 
                 // We copied everything up to, but not including, the colon:
                 assert(ifconfig_line[dst_length] == ':');
                 assert(current_interface[dst_length] == '\0');
+#endif // not NDEBUG
```
Comment (larsewi) [inline libenv/unix_iface.c:722]: Use this macro instead
```suggestion
                // We know there was more data, at least a colon:
                assert(src_length == bytes_to_copy);
                NDEBUG_UNUSED const size_t dst_length = src_length - 1;

                // We copied everything up to, but not including, the colon:
                assert(ifconfig_line[dst_length] == ':');
                assert(current_interface[dst_length] == '\0');
```

**Review by larsewi — COMMENTED** [review]
🚀

File: libenv/unix_iface.c (hunk 2 above)
Comment (craigcomstock) [reply libenv/unix_iface.c:722]: That looks great but how would I have found that? ;) Hmm...

**Comment by craigcomstock** [comment]
will wait on https://github.com/cfengine/libntech/pull/175, when that is merged, I will update/squash and open for review/merge.

**Comment by craigcomstock** [comment]
@cf-bottom jenkins

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/9285/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/9285/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-9285/

**Comment by craigcomstock** [comment]
@cf-bottom jenkins please, thanks.

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/9287/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/9287/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-9287/

**Comment by craigcomstock** [comment]
@cf-bottom jenkins after removing stale libntech pr

**Review by olehermanse — APPROVED** [review]
LGTM thanks (just wait to see it green-ish in Jenkins) :)

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/9288/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/9288/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-9288/

**Review by larsewi — APPROVED** [review]

**Comment by craigcomstock** [comment]
ci was flaky ssh on sequential-tests and red for selenium, so merged. :)

**Comment by craigcomstock** [comment]
cherry picks:
3.18.x - https://github.com/cfengine/core/pull/5185
3.21.x - https://github.com/cfengine/core/pull/5183

**Pull request merged**